### ISSUE TITLE
`$_AwaitThenValue1` is not defined

### ISSUE THREAD

**Issue by marekdedic** [ota-meshi/eslint-plugin-svelte#335]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v8.23.0

### What version of `eslint-plugin-svelte` are you using?

2.14.0

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "parser": "svelte-eslint-parser",
  "parserOptions": {
    "parser": "@typescript-eslint/parser",
    "extraFileExtensions": [".svelte"]
  },
  "rules": {
    "no-undef": "error"
  }
}
```
</details>

```svelte
<script lang="ts">
  const contentPromise = new Promise();
</script>

{#if false}
{:else if true}
  {#await contentPromise then _}
  {/await}
{/if}
```

### What did you expect to happen?

Either no error, or an error message that I can understand

### What actually happened?

```
/home/marekdedic/eslint-svelte-repro/Error.svelte
  8:3  error  '$_AwaitThenValue1' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://gist.github.com/marekdedic/7394ecd403fdcda3c516a7ea76c944be

**Comment by ota-meshi** [comment]
Duplicate of #325.
I think updating the parser will fix the problem.

**Comment by marekdedic** [comment]
You are correct, sorry for the duplicate.
